vcstool/clients/vcs_base.py

```diff
@@ -93,7 +113,12 @@ def load_url(url, retry=2, retry_period=1, timeout=10):
     try:
         fh = urlopen(url, timeout=timeout)
     except HTTPError as e:
-        if e.code == 503 and retry:
+        if e.code in (401, 404):
+            # Try again, but with authentication
+            fh = _authenticated_urlopen(url, timeout=timeout)
+            if fh is not None:
+                return fh.read()
+        elif e.code == 503 and retry:
```

Comment:
In the case of a retry the next iteration would again first try a `urlopen` just to then use `_authenticated_urlopen` again. It would be better if a retry will immediately use the authenticated version if it was necessary in the first iteration.

Reply:
Actually if it hits this branch of code it never hit `_authenticated_urlopen`, but your point is still valid: `_authenticated_urlopen()` doesn't support the retry logic, and it should indeed. I'll add it.

Reply:
Fixed in b49456971293af1141f9df1d0f051a32b4107fec.